src/server/job_func.c

```diff
@@ -416,6 +416,57 @@ job_alloc(void)
 	return (pj);
 }
 
+#ifndef PBS_MOM
+
+/**
+ * @brief
+ * 	free work tasks and pending batch requests related to this job
+ *
+ * @param[in]	pj - pointer to job structure
+ *
+ * @return	void
+ */
+void
+free_job_work_tasks(job *pj)
+{
+	struct work_task *pwt;
+	struct batch_request *tbr = NULL;
+	/*
+	* Delete any work task entries associated with the job.
+	* mom deferred tasks via TPP are also hooked into the
+	* ji_svrtask now, so they also get automatically cleared
+	* in this following loop
+	*/
+	while ((pwt = (struct work_task *)GET_NEXT(pj->ji_svrtask)) != NULL) {
+		if (pwt->wt_type == WORK_Deferred_Reply) {
+			tbr = (struct batch_request *)pwt->wt_parm1;
+			if (tbr != NULL) {
+				/* Check if the reply is for scheduler
+					* If so, then reject the request.
+					*/
+				if ((tbr->rq_orgconn != -1) &&
+					(find_sched_from_sock(tbr->rq_orgconn) != NULL)) {
+					tbr->rq_conn = tbr->rq_orgconn;
+					req_reject(PBSE_HISTJOBID, 0, tbr);
```

Comment:
Is HISTJOBID the right reject code here?  Job history might not be on.

Reply:
This was the same error code used in job_free() which is called when history is disabled. 
I will modify the code to return HISTJOBID in case of history enabled and UNKNOWN job id in case of no history. Thanks!

Reply:
Actually, on second thoughts, I think i would like to leave it as is. This is the exact same code that was being used in the case of history-disabled, right, so it must have been working - or the callers do not care. In other words, i think tampering the return code is probably more dangerous than leaving it as is. 

The change here is returning an error in the case of history jobs and for that the return code is appropriate anyway.